sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -858,17 +928,376 @@ block_load(const T *ptr, simd_mask<1> pred, simd<T, N> old_values,
+/// The parameter 'props' specifies the optional compile-time properties
+/// of the type esimd::properties and may include esimd::cache_hint_L1,
+/// esimd::cache_hint_L2, esimd::cache_hint_L3, esimd::alignment.
+
+/// simd<T, N>
+/// block_load(AccessorT acc, OffsetT offset, props = {});         // (acc-1)
+/// simd<T, N> block_load(AccessorT acc, props);                   // (acc-2)
+
+/// simd<T, N>
+/// block_load(AccessorT acc, OffsetT offset, simd_mask<1> pred,
+///            simd<T, N> pass_thru, flags = {});                  // (acc-3)
+/// simd<T, N>
+/// block_load(AccessorT acc, OffsetT offset, simd_mask<1> pred,
+///            flags = {});                                        // (acc-4)
+
+/// simd<T, N>
+/// block_load(AccessorT acc, simd_mask<1> pred,
+///            simd<T, N> pass_thru, flags = {});                  // (acc-5)
+/// simd<T, N>
+/// block_load(AccessorT acc, simd_mask<1> pred, flags = {});      // (acc-6)
+
+/// simd<T, N>
+/// block_load(AccessorT acc, OffsetT offset, props = {});         // (acc-1)
+/// This function loads a contiguous memory block referenced
+/// by accessor \p acc and byte-offset \p offset.
+///
+/// The parameter \p props specifies the optional compile-time properties
+/// of the type esimd::properties and may include esimd::cache_hint_L1,
+/// esimd::cache_hint_L2, esimd::alignment. Other properties are ignored.
+///
+/// Cache hints: If \p props does not specify any L1 or L2 cache hints, then
+/// the cache_hint::none value is assumed by default.
+///
+/// Alignment: If \p props does not specify the 'alignment' property, then
+/// the \p offset must be at least 4-byte aligned for elements of 4-bytes or
+///     smaller and 8-byte aligned for 8-byte elements.
+///
+/// Restrictions - cache hint imposed - temporary:
+/// If L1 or L2 cache hint is passed, then:
+/// R1: The pointer must be at least 4-byte aligned for elements of 4-bytes or
+///     smaller and 8-byte aligned for 8-byte elements.
+/// R2: The number of elements for 8-byte data: 1, 2, 3, 4, 8, 16, 32, 64;
+///     for 4-byte data: 1, 2, 3, 4, 8, 16, 32, 64,
+///                      or 128(only if alignment is 8-bytes or more);
+///     for 2-byte data: 2, 4, 6, 8, 16, 32, 64, 128,
+///                      or 256(only if alignment is 8-bytes or more);
+///     for 1-byte data: 4, 8, 12, 16, 32, 64, 128, 256,
+///                      or 512(only if alignment is 8-bytes or more).
+/// R3: The target device must be DG2, PVC or newer GPU.
+/// (R1), (R2) and (R3) are not applied if there are no cache hints.
+template <typename T, int N, typename AccessorT,
+          typename PropertyListT =
+              ext::oneapi::experimental::detail::empty_properties_t>
+__ESIMD_API std::enable_if_t<
+    ext::oneapi::experimental::is_property_list_v<PropertyListT> &&
+        sycl::detail::acc_properties::is_accessor_v<AccessorT> &&
+        !sycl::detail::acc_properties::is_local_accessor_v<AccessorT>,
```

Comment:
Maybe a dumb question but we don't allow local accessors for a couple of the functions here, why is that?

Reply:
There is such function in experimental/memory.hpp. It is just not ported by this patch yet:
```
template <typename T, int NElts, lsc_data_size DS = lsc_data_size::default_size,
          cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
          typename AccessorTy,
          typename FlagsT = __ESIMD_DNS::dqword_element_aligned_tag>
__ESIMD_API std::enable_if_t<
    sycl::detail::acc_properties::is_local_accessor_v<AccessorTy> &&
        __ESIMD_NS::is_simd_flag_type_v<FlagsT>,
    __ESIMD_NS::simd<T, NElts>>
lsc_block_load(AccessorTy acc, uint32_t offset,
               __ESIMD_NS::simd_mask<1> pred = 1, FlagsT flags = FlagsT{}) {
  return lsc_slm_block_load<T, NElts, DS>(
      offset + __ESIMD_DNS::localAccessorToOffset(acc), pred);
}
```